vscode/src/test/suite/testutils.ts

```diff
@@ -192,3 +192,278 @@ export async function dumpJava(): Promise<void> {
+    const propertiesLocalisableFields = ["description", "enumDescriptions"];
+    let property: any;
+    for (const propertyId of configPropertiesIds) {
+        property = configuration.properties[propertyId];
+        if (!isLocalizedObj(property, propertiesLocalisableFields, propertyId, "Configuration Property", validKeys)) localized = false;
+    }
+    return localized;
+}
+
+export function checkDebuggersLocalisation(debuggers: any, validKeys: Set<string>): boolean {
+    let localized = true;
+    const propertiesLocalisableFields = ['description'];
+    const configLocalisableFields = ['name'];
+    const snippetLocalisableFields = ['label', 'description'];
+    for (const debug of debuggers) {
+        // check configurationAttributes
+        for (const [propName, prop] of Object.entries(debug.configurationAttributes.launch.properties)) {
+            if (!isLocalizedObj(prop, propertiesLocalisableFields, propName, "Configuration Attributes : Launch properties ", validKeys)) localized = false;
+        }
+        // check initialConfigurations
+        for (const initConfig of debug.initialConfigurations) {
+            if (!isLocalizedObj(initConfig, configLocalisableFields, "", "Initial Configuration", validKeys)) localized = false;
+        }
+        // check configurationSnippets
+        for (const snippet of debug.configurationSnippets) {
+            if (!isLocalizedObj(snippet, snippetLocalisableFields, "", "configuration Snippet", validKeys)) localized = false;
+        }
+    }
+    return localized;
+}
+
+
+export function checkViewsLocalisation(views: any, validKeys: Set<string>): boolean {
+    let localized: boolean = true;
+    const explorer: any = views.explorer;
+    const explorerLocalisableFields = ["name", "contextualTitle"];
+    let localizingKey: string;
+    var fieldVal: string;
+    var id: string;
+    for (const obj of explorer) {
+        if (!isLocalizedObj(obj, explorerLocalisableFields, obj.id, "explorer", validKeys)) localized = false;
+    }
+    return localized;
+}
+
+
+export function checkCommandsLocalisation(commands: any, validKeys: Set<string>): boolean {
+    const localisableFields = ['title'];
+    let localized: boolean = true;
+    let localizingKey: string;
+    var fieldVal: any;
+    var id: string;
+    for (const command of commands) {
+        id = command.command;
+        for (const field of localisableFields) {
+            fieldVal = command[field];
+            if (fieldVal === undefined) continue;
+            localizingKey = getlocalizingKey(fieldVal);
+            if (!isLocalizedVal(fieldVal)) {
+                localized = false;
+                console.log(`${field} of the command '${id}' is not being localized`)
+            } else if (!validKeys.has(localizingKey)) {
+                console.log(`Invalid localizing key for the command '${id}' key '${localizingKey}' not found in the package.nls.json`);
+            }
+        }
```

Comment:
Could this be refactored to re-use the same code as in `isLocalizedObj()`? That would take care of the array out-of-bounds err mentioned there.

Reply:
Sure !